### PR TITLE
[expo-yarn-workspaces]Fix Windows compatibility issues

### PR DESCRIPTION
**Pull request by nattyrice**
# Why

A fix for several Windows compatibility problems.  One problem is detailed [here](https://github.com/expo/expo/issues/4215)
A second compatibility problem became apparent once I addressed the initial `spawnSync UNKNOWN` problem.  This problem consisted of a `spawnSync ENOENT` error.  It occurred because Windows expects commands to be postfixed with `.cmd`  e.g. `yarn.cmd`

I also addressed a slight style inconsistency in one of the scripts.  It had named local helper functions as if they were private variables(prefixed with _).

# How
I addressed both issues by changing all the child_process.execFileSync calls to spawnSync calls wrapped in a function that checks the platform and adjusts the call as needed.

# Test Plan
I tested this fix on a typical yarn workspace setup to use an expo and expo-yarn-workspaces package.
I edited the files in my `node_modules` folder directly.  I set up package.json scripts to test the package. 
As an addition test I ran `npx expo-yarn-workspaces postinstall` or `npx expo-yarn-workspaces check-workspace-dependencies` manually.

### SpawnSync UNKNOWN Error
image

### SpawnSync ENOENT Error
image

### Package.json
image

### Script Test
image

### NPX Test
image

### CODE REVIEW

File: packages/expo-yarn-workspaces/bin/expo-yarn-workspaces.js [hunk 1]
```diff
@@ -1,25 +1,26 @@
 #!/usr/bin/env node
-'use strict';
+"use strict";
```
Comment (ide) [inline packages/expo-yarn-workspaces/bin/expo-yarn-workspaces.js]: Run Prettier over these files

File: packages/expo-yarn-workspaces/common/cross-spawn-sync.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const os = require("os").platform();
+
+let selectiveCmdify = cmd => cmd;
+if (os === "win32") {
+  selectiveCmdify = cmd => cmd + ".cmd";
+}
+
+module.exports = function spawnSync(cmd, args, options) {
+  cmd = selectiveCmdify(cmd);
+  if (!Array.isArray(args)) {
```
Comment (ide) [inline packages/expo-yarn-workspaces/common/cross-spawn-sync.js]: remove this, make it the caller's responsibility to pass in an array

File: packages/expo-yarn-workspaces/common/cross-spawn-sync.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+"use strict";
+
+const child_process = require("child_process");
+const os = require("os").platform();
+
+let selectiveCmdify = cmd => cmd;
```
Comment (ide) [inline packages/expo-yarn-workspaces/common/cross-spawn-sync.js]: inline this logic in `spawnSync`, easier to read with less indirection

**Review by ide — CHANGES_REQUESTED** [review]

File: packages/expo-yarn-workspaces/bin/expo-yarn-workspaces.js (hunk 1 above)
Comment (nattyrice) [reply packages/expo-yarn-workspaces/bin/expo-yarn-workspaces.js]: Done

File: packages/expo-yarn-workspaces/common/cross-spawn-sync.js (hunk 2 above)
Comment (nattyrice) [reply packages/expo-yarn-workspaces/common/cross-spawn-sync.js]: done

File: packages/expo-yarn-workspaces/common/cross-spawn-sync.js (hunk 3 above)
Comment (nattyrice) [reply packages/expo-yarn-workspaces/common/cross-spawn-sync.js]: done

File: packages/expo-yarn-workspaces/common/cross-spawn-sync.js (hunk 2 above)
Comment (nattyrice) [reply packages/expo-yarn-workspaces/common/cross-spawn-sync.js]: One sec

Comment (nattyrice) [reply packages/expo-yarn-workspaces/common/cross-spawn-sync.js]: Now, it is done

**Review by ide — APPROVED** [review]

**Comment by brownbl1** [comment]
I'm running into this as well. Thanks for the PR! @ide do you know what version this will be released with?

**Pull request merged**